### PR TITLE
Aggregation: Fix squished aggregation chart

### PR DESCRIPTION
**Pull request by vovakulikov**
Closes https://github.com/sourcegraph/sourcegraph/issues/41158
## Background
We know now that there is some strange behaviour about empty line measurement in SVG. If SVG text has no child or text content then measurement is corrupted and it breaks the chart layout calculation. This bug doesn't exist in Firefox for example, in Safari it appears but chart isn't squished completely (just a little) and only Chrome breaks the chart entirely.

This PR replaces empty SVG text elements textContent with whitespaces. This fixes problem in all three major browsers. 

## Test plan
- Run this query ` query(.*)string repo:^github\.com/sourcegraph/sourcegraph$ ` on sourcegraph.sourcegraph.com staging (or locally) 
- Make sure that chart looks fine with groups that returns this query. 

## App preview:

- [Web](https://sg-web-vk-fix-squished-aggregation-chart.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-vfeiwaixzx.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/search/results/components/aggregation/AggregationChart.tsx [hunk 1]
```diff
@@ -84,7 +84,7 @@ const getXScaleTicks = <T,>(options: GetScaleTicksOptions): T[] => {
         filteredTicks = filteredTicks.filter((tick, index) => index % 2 === 0)
     }
 
-    return filteredTicks
+    return filteredTicks.map(tick => tick.trim()).filter(tick => !!tick)
```
Comment (chwarwick) [inline client/web/src/search/results/components/aggregation/AggregationChart.tsx:87]: Is it possible you will end up with no ticks or unbalanced ticks this way?

Comment (unclejustin) [reply client/web/src/search/results/components/aggregation/AggregationChart.tsx:87]: Same question. Could we add a story to storybook that resembles the problem?

Comment (vovakulikov) [reply client/web/src/search/results/components/aggregation/AggregationChart.tsx:87]: This is absolutely possible

Comment (vovakulikov) [reply client/web/src/search/results/components/aggregation/AggregationChart.tsx:87]: @unclejustin I've added the story with bad data that breaks the chart on the main

**Comment by vovakulikov** [comment]
@unclejustin @chwarwick I think I found a different solution for this problem that doesn't imply skipping any labels from the chart. Take a look.

File: client/web/src/search/results/components/aggregation/AggregationChartCard.tsx [hunk 2]
```diff
@@ -20,8 +20,8 @@ const LazyAggregationChart = lazyComponent<AggregationChartProps<SearchAggregati
 
 const getName = (datum: SearchAggregationDatum): string => datum.label ?? ''
 const getValue = (datum: SearchAggregationDatum): number => datum.count
-const getColor = (datum: SearchAggregationDatum): string => (datum.label ? 'var(--primary)' : 'var(--text-muted)')
 const getLink = (datum: SearchAggregationDatum): string => datum.query ?? ''
+const getColor = (): string => 'var(--primary)'
```
Comment (valerybugakov) [inline client/web/src/search/results/components/aggregation/AggregationChartCard.tsx:24]: It's not clear what's the reason for this change from the PR description.

File: client/wildcard/src/components/Charts/core/components/axis/Tick.tsx [hunk 3]
```diff
@@ -13,7 +13,22 @@ export interface TickProps extends TickRendererProps {
...
+    const {
+        children,
+        formattedValue = '',
+        className,
+        'aria-label': ariaLabel,
+        getTruncatedTick,
+        ...tickLabelProps
+    } = props
+
+    // Formatted and truanted tick value
```
Comment (valerybugakov) [inline client/wildcard/src/components/Charts/core/components/axis/Tick.tsx:25]:
```suggestion
    // Formatted and truncated tick value
```

**Review by valerybugakov — APPROVED** [review]
Tested using the updated Storybook story in major browsers ✅

File: client/web/src/search/results/components/aggregation/AggregationChartCard.tsx (hunk 2 above)
Comment (vovakulikov) [reply client/web/src/search/results/components/aggregation/AggregationChartCard.tsx:24]: Yeah, this is something that I noted during this PR, it isn't a part of the fix for the chart layout
